### DISCUSSION TITLE
How can I read a large csv file by splitting it into chunks?

### DISCUSSION THREAD

**Discussion by antoniofrs** [hosseinmoein/DataFrame#228] [General]
I'm having trouble looking for a solution that allows me to read a very large csv file.

Something like this, but in c++ 🙄
```python
for chunk in pd.read_csv(filename, chunksize=chunksize):
    process(chunk)
```

Thanks in advance!

**Comment by hosseinmoein** [comment]
Yes. You can use the `read()` function of DataFrame. The docs and code samples are here https://hosseinmoein.github.io/DataFrame/docs/HTML/DataFrame.html.
You should read the short doc to get familiar with the general usage of DataFrame.

To read csv files with DataFrame you need the `io_format::csv2` option. You also need to modify the header -- first line of the csv file -- to conform with the format that DataFrame expects.
